pkgs/applications/misc/nemu/default.nix

```diff
@@ -0,0 +1,65 @@
+{ stdenv, lib, fetchFromGitHub
+, cmake, pkgconfig, gettext
+, ncurses, sqlite, libusb, udev
+
+, libxml2 ? null
+, libarchive ? null
+, graphviz ? null
+, dbus ? null
+
+, debugBuild ? false
+, ovfSupport ? true
+, saveVMSupport ? false
+, withVNCClientSupport ? true
+, withSpice ? true
+, withNetworkMap ? false
+, withDbus ? false
+}:
+
+assert ovfSupport     -> libxml2    != null;
+assert ovfSupport     -> libarchive != null;
+assert withNetworkMap -> graphviz   != null;
+assert withDbus       -> dbus       != null;
+
+stdenv.mkDerivation rec {
+  pname = "nemu";
+  version = "2.3.0";
+
+  src = fetchFromGitHub {
+    owner = "nemuTUI";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wyps72w59mh3bjx533qy88wypyg95gwkmyf7p67xxdlfwbjgvv5";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig gettext ];
+
+  buildInputs = [ ncurses sqlite libusb udev libxml2 ]
+    ++ (lib.optional ovfSupport     libarchive)
+    ++ (lib.optional withNetworkMap graphviz)
+    ++ (lib.optional withDbus       dbus);
+
+  cmakeFlags = []
+    ++ (lib.optional debugBuild "-DNM_DEBUG=ON")
+    ++ (lib.optional ovfSupport "-DNM_WITH_OVF_SUPPORT=ON")
+    ++ (lib.optional saveVMSupport "-DNM_SAVEVM_SNAPSHOTS=ON")
+    ++ (lib.optional withVNCClientSupport "-DNM_WITH_VNC_CLIENT=ON")
+    ++ (lib.optional withSpice "-DNM_WITH_SPICE=ON")
+    ++ (lib.optional withNetworkMap "-DNM_WITH_NETWORK_MAP=ON")
+    ++ (lib.optional withDbus "-DNM_WITH_DBUS=ON")
+    ;
+
+  installPhase = ''
```

Comment:
Upstream doesn't know how to write `CMakeLists.txt`. The following warning messages in `configurePhase` indicate this:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_CXX_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
```

And it's notable when reading this:

https://github.com/nemuTUI/nemu/blob/1633a9ecb2b9b036a01a5dbeddb21787680064ec/CMakeLists.txt#L24-L36

I say - report this upstream and once fixed, use a `fetchpatch` remove the `installPhase` override.

Reply:
In particular, most of these are caused not using GNUInstallDirs cmake module.

Reply:
We updated our CMakeLists.txt
https://github.com/nemuTUI/nemu/commit/a3f179abd62ed23ccb8de2f79820aee2791767cf

Can you guys check if it's ok in current state, if so, we can make a new release for your package

Reply:
Evaluation has failed: https://gist.github.com/f2bfb5498488aa5d2895bcbcace14e8d because of `libudev` argument not suppliled.

Reply:
@matthiasbeyer please fix the inputs.